### ISSUE TITLE
[1.20.1] Applied Botanics 1.5.0 keeps thinking there is an update when there isn't

### ISSUE THREAD

**Issue by jamieb999** [ramidzkh/Applied-Botanics#61]
So when I starte the game and I go into the mods it keeps thinking that
there is an update for 1.5.0 of Applied Botanics but there is no update
available i go the website by clicking on the website button and 
it takes me to to the website. I have tried version from there
and curseforge and they are the same and it keeps saying
there is an update.

The site it takes me to is at 
https://modrinth.com/mod/applied-botanics

https://modrinth.com/mod/applied-botanics/version/1.5.0-forge
I have that version already. 

Also tried the verison from Curseforge. 
https://www.curseforge.com/minecraft/mc-mods/applied-botanics-addon

Deleted the mod and reinstalled it two on both serverside and client
side and yet still it thinks there is an update when I am running 1.5.0
which according to both sites is the latest verison. 

Still the same issue says there is an update when there is not.
I check and I am running the latest version.  So why won't it go
away in my update list. 

https://www.curseforge.com/minecraft/mc-mods/applied-botanics-addon/files/4904185

Running 
1.20.1 Neoforge  forge-1.20.1-47.1.91

Can you please fix your mod why does it keep thinking there is an update
when there isn't. 

And don't try and blame the update checker or something like that because it
only does this with this ONE mod I have over 150 mods installed and 
it only does this with Applied Botanics 1.5.0. 

This is an issue with the mod itself I believe. 

This is even coming up in my logs server side as well 

[27Jan.2024 22:02:45.115] [Forge Version Check/INFO] [net.minecraftforge.fml.VersionChecker/]: [appbot] Starting version check at https:                                           //api.modrinth.com/updates/applied-botanics/forge_updates.json
{"homepage":"https://modrinth.com/mod/applied-botanics","promos":{"1.20.1-latest":"1.5.0-forge","1.19.2-latest":"1.4.6-forge","1.19.2-re                                           commended":"1.4.6-forge","1.20.1-recommended":"1.5.0-forge"}}

I have 1.5.0 and yet it still thinks client side there is an update when there isn't I have the latest
version.  So why it is doing this only with Applied Botanics and not the over 150 other mods I have 
installed on this server. It is just this one. 

![2024-01-29_14 07 04](https://github.com/ramidzkh/Applied-Botanics/assets/115375746/895f0ab7-4e77-4865-97ce-677f4a6aa835)

![2024-01-29_14 07 13](https://github.com/ramidzkh/Applied-Botanics/assets/115375746/f5018642-a3a0-481a-82e4-0adb024e822d)

**Comment by Mari023** [comment]
this happens because the version is named "1.5.0-forge" on modrinth, but "1.5.0" ingame

**Comment by jamieb999** [comment]
Well it needs to be fixed.  This is an issue with the mod and needs to be fixed. 
I didn't change the name at all I downloaded it as is.  So this is something
the developers need to look at and resolve. 

https://www.curseforge.com/minecraft/mc-mods/applied-botanics-addon/files/4904185

When you download this from Curse Forge the file name is 
Applied-Botanics-forge-1.5.0.jar 

As I said I did not change the name or anything and I am using it as is. 
It still thinks that there is an update for 1.5.0 when there isn't.

This needs to be looked at by the dev please!

**Comment by jamieb999** [comment]
Ok so an update on this.. from what  Mari023 said

"this happens because the version is named "1.5.0-forge" on modrinth, but "1.5.0" ingame"

So I tried this.. I renamed the file to Applied-Botanics-1.5.0.jar

So I removed the  "forge" from the name because this person said that
it was the cause of it thinking there is an update. 

It is the same on both the server and the client and that still doesn't fix the
issue it is still saying there is an update for 1.5.0  even after doing this
so it does not fix the issue and 1.5.0 is the latest version
of Applied Botanics as well still at the time.

So stopping the server and renaming the file on both client and server so it 
is listed as above does not fix this problem it is still saying there is an update
for 1.5.0 when you look under the mods when you are starting up Minecraft. 

So this does not fix the issue so there is still an issue here this was tested
on March 1 2024 about 7:00 PM MST (GMT-0700) 

This was downloaded as well from Curseforge 

https://www.curseforge.com/minecraft/mc-mods/applied-botanics-addon
https://www.curseforge.com/minecraft/mc-mods/applied-botanics-addon/files/4904185

So this problem still persists with it thinking that there is an update with 1.5.0 
when it is the latest version this is most definately a bug. 

![image](https://github.com/ramidzkh/Applied-Botanics/assets/115375746/c6395515-13dd-4ef7-af32-5ffbf9e3b0b8)

**Comment by Mari023** [comment]
> So I tried this.. I renamed the file to Applied-Botanics-1.5.0.jar

that doesn't work, the file name doesn't affect anything, it only has to be unique in the folder

If you really want to fix this by editing the mod file, here is how to do it:
1. a jar is just a .zip file with a different file ending, you can open it like a zip file (with some programs you have to rename it to .zip to edit it (don't forget to rename it back afterwards))
2. in the zip folder there is a folder called `META-INF`, containing a file called `mods.toml`
3. edit that file, you can either:
- remove the line `updateJSONURL = "https://api.modrinth.com/updates/applied-botanics/forge_updates.json"`, which will disable the version checking for Applied-Botanics
- or change `version = "1.5.0"` to `version = "1.5.0-forge"`, which will make it correctly detect that version as the latest version

**Comment by jamieb999** [comment]
Yes it should have worked even by what you said did you not READ what 
I did. 

Read what you said 

"Or change version = "1.5.0""

It does not work! 

I shouldn't have to do this the creator of the mod needs to fix this. This is NOT working
have the creator of this mod fix it and do 1.5.1

This is a bug  This is garbage where is the mod for this mod fix it this
is a bug with the mod! 

Fix this mod immediately!

**Comment by castilio20** [comment]
Well you have an IT degree, Pr a fix in then since you know it all princess

**Comment by jamieb999** [comment]
Wow you report something and this is how you get treated with comments like castilio20. 
This is a BUG with the mod and the dev seems to refuse to fix it.  Wow just horrible! Fix the bug 
immediately!

**Comment by castilio20** [comment]
you reported a bug congrats you want more nice comments be more nice also it a fck minor bug it nothing rlly that big so if they fix this doesnt rlly mater it not like it crash the game or somthing the dev could use the time to do better things  that fixing that

**Comment by Xplodin** [comment]
> Wow you report something and this is how you get treated with comments like castilio20. This is a BUG with the mod and the dev seems to refuse to fix it. Wow just horrible! Fix the bug immediately!

fix yourself coward.
Developers arent slaves.

**Comment by jamieb999** [comment]
Well it has been over 3 months since I reported this and it still isn't fixed I am not saying the developers are
"slaves" it is like 3.5 months now since i reported this and the issue still isn't fixed please release 1.50.1 and fix this 
bug please developer.  For the record this mod hasn't been updated since Nov 26, 2023 is this mod dead? is the
developer not developing it any more? 

 I do appreciate your work FYI to the developer.
